April Serenity updates (#589)
akeneo/pim-helpcenter: content/updates/04-2020/4-rules-engine-new-actions-and-improvements.md

# Rules engine new actions and improvements

## Calculate attribute values

> ::: meta-data type="New" features="Rules engine, Productivity, Automation" available="9th of April" link-to-doc="../articles/get-started-with-the-rules-engine.html#calculate"

Reviewer: ```suggestion
::: meta-data type="New" features="Rules engine, Productivity, Automation" available="early April" link-to-doc="../articles/get-started-with-the-rules-engine.html#calculate"
```

:::

> **From complexed to simple calculations, the `Calculate` rule action enables you to do mathematical operations to fill in attributes based on other attribute values**. Multiply your length, width and height attributes to get a volume, or create a more complex calculation to forecast your margin.

Reviewer 2: ```suggestion
**From complex to simple calculations, the `Calculate` rule action enables you to do mathematical operations to fill in attributes based on other attribute values**. Multiply your length, width and height attributes to get a volume, or create a more complex calculation to forecast your margin.
```

> **No more time wasting calculating values, the `Calculate` action does it for you.**

Reviewer 2: ```suggestion
**No more time wasted calculating values, the `Calculate` action does it for you.**
```

## Empty product values

> ::: meta-data type="New" features="Rules engine, Productivity, Automation" available="9th of April" link-to-doc="../articles/get-started-with-the-rules-engine.html#clear"

Reviewer: ```suggestion
::: meta-data type="New" features="Rules engine, Productivity, Automation" available="early April" link-to-doc="../articles/get-started-with-the-rules-engine.html#clear"
```

:::

> **Easily empty attribute values thanks to the new `Clear` rule action!** Automatically remove value from any attribute: a date, a price, a simple-select.. or any other field based on specific conditions. This new action also allows you to remove categories from a product, or ungroup/disassociate products from others automatically.

Reviewer 2: ```suggestion
**Easily empty attribute values thanks to the new `Clear` rule action!** Automatically remove values from any attribute: a date, a price, a simple-select.. or any other field based on specific conditions. This new action also allows you to remove categories from a product, or ungroup/disassociate products from others automatically.
```

## Name your rules

> ::: meta-data type="New" features="Rules engine, Productivity, Automation" available="9th of April" link-to-doc="../articles/get-started-with-the-rules-engine.html#name-your-rules"

Reviewer: ```suggestion
::: meta-data type="New" features="Rules engine, Productivity, Automation" available="early April" link-to-doc="../articles/get-started-with-the-rules-engine.html#name-your-rules"
```

:::

You can now **name your rules with a clear label**. And that's not it, you can define a rule name per PIM locale enabled. The rule's label is visible under your rules grid and under your smart attributes, if you have international teams you can now make sure that they understand the purpose of a rule.

> ## Set or add associations actions improved

Reviewer 2: ```suggestion
## Set or add association actions improved
```

> ::: meta-data type="Improvement" features="Rules engine, Productivity, Automation" available="9th of April" link-to-doc="../articles/get-started-with-the-rules-engine.html#add-associations"

Reviewer: ```suggestion
::: meta-data type="Improvement" features="Rules engine, Productivity, Automation" available="early April" link-to-doc="../articles/get-started-with-the-rules-engine.html#add-associations"
```

:::

> It was already possible to automatically set or add associations to products/product models/groups with the rules engine, but the format was a bit complex. **We simplified the format so it is easier for you to use those actions.**

Reviewer 2: ```suggestion
It was already possible to automatically set or add associations to products/product models/groups with the rules engine, but the format was a bit complex. **We simplified the format so that it is easier for you to use those actions.**
```

::: more

> [Get started with the rules engine](../articles/get-started-with-the-rules-engine.html)   

Reviewer: ```suggestion
[What is the rules engine?](../articles/get-started-with-the-rules-engine.html)   
```

> [Manage your rules](../articles/manage-your-rules.html)

Reviewer: ```suggestion
[How to manage my rules](../articles/manage-your-rules.html)
```
